framework/resource_router.go

```diff
@@ -1,9 +1,110 @@
 package framework
 
-import "context"
+import (
+	"context"
+	"fmt"
 
-func DefaultResourceRouter(rs []Resource) func(ctx context.Context, obj interface{}) ([]Resource, error) {
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/micrologger/loggermeta"
+	"k8s.io/apimachinery/pkg/api/meta"
+
+	"github.com/giantswarm/operatorkit/framework/context/reconciliationcanceledcontext"
+	"github.com/giantswarm/operatorkit/framework/context/resourcecanceledcontext"
+	"github.com/giantswarm/operatorkit/framework/context/updateallowedcontext"
+	"github.com/giantswarm/operatorkit/framework/context/updatenecessarycontext"
+)
+
+type ResourceRouterConfig struct {
+	CustomObjectVersionFunc func(obj interface{}) (string, error)
+	CtxFunc                 func(ctx context.Context, obj interface{}) (context.Context, error)
```

Comment:
I think `Init` prefix we had (i.e. `InitCtxFunc`) was a good thing. It was indicating that this func initializes the context.